EpiAware/test/EpiLatentModels/models/RandomWalk.jl

```diff
@@ -38,3 +38,39 @@ end
     @test rw_process.init_prior == init_prior
     @test rw_process.std_prior == std_prior
 end
+
+@testitem "Testing RandomWalk parameter recovery: Negative Binomial errors on log rw" begin
+    using Random, Turing, FillArrays, Distributions, LinearAlgebra, DynamicPPL, StatsBase,
+          ReverseDiff
+    Random.seed!(1234)
+
+    rw_process = RandomWalk()
+    obs_nb = NegativeBinomialError()
+
+    @model function test_negbin_errors(rw, obs, y_t)
+        n = length(y_t)
+        @submodel Z_t = generate_latent(rw, n)
+        @submodel y_t = generate_observations(obs, y_t, exp.(Z_t))
+        return Z_t, y_t
+    end
+
+    generative_mdl = test_negbin_errors(rw_process, obs_nb, fill(missing, 40))
+    θ_true = rand(generative_mdl)
+    Z_t_obs, y_t_obs = condition(generative_mdl, θ_true)()
+
+    mdl = test_negbin_errors(rw_process, obs_nb, Int.(y_t_obs))
+    chn = sample(
+        mdl, NUTS(adtype = AutoReverseDiff(; compile = Val(true))), 1000, progess = false)
+
+    #Check that are in central 99.9% of the posterior predictive distribution
```

Comment:
we could maybe make this a bit higher i.e 99%

Reply:
This is where I'm a bit torn. If we have a test on 99% cred. int. then assuming the posterior dist has accurate coverage then if we test 10 params we expect it to pass $0.99^10 \approx 0.9$ proportion of the time. But a one in 10 stochastic CI failure rate could be a bit annoying?

Otoh, I set random seed so if we found one that worked (for Mac, Window and Ubuntu) then it will work until we do something that breaks it.

I don't have a lot of experience with optimizing an inference package so not sure where to pitch this.

Reply:
but we are fixing seeds right and also I assume the intervals aren't independent?

Reply:
> but we are fixing seeds right and also I assume the intervals aren't independent?

Good point... failure is probably pretty strongly correlated here.

Reply:
Changed default to 0.99.